File: src/main/java/net/hellomouse/xeno_early_start/registries/ProgressionModLootTypeRegistry.java
package net.hellomouse.xeno_early_start.registries;

import net.hellomouse.xeno_early_start.ProgressionMod;
import net.hellomouse.xeno_early_start.loot.ConfigLootFunction;
import net.hellomouse.xeno_early_start.loot.PickaxeTier;
import net.minecraft.loot.condition.LootConditionType;
import net.minecraft.loot.function.LootFunctionType;
import net.minecraft.registry.Registries;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.RegistryObject;

public class ProgressionModLootTypeRegistry {
    public static final DeferredRegister<LootConditionType> COND_DEF_REG = DeferredRegister.create(Registries.LOOT_CONDITION_TYPE.getKey(), ProgressionMod.MODID);
    public static final RegistryObject<LootConditionType> PickaxeTier = COND_DEF_REG.register("pickaxe_tier", () -> new LootConditionType(new PickaxeTier.Serializer()));
    public static final DeferredRegister<LootFunctionType> FUNC_DEF_REG = DeferredRegister.create(Registries.LOOT_FUNCTION_TYPE.getKey(), ProgressionMod.MODID);
    public static final RegistryObject<LootFunctionType> ConfigLootFunction = FUNC_DEF_REG.register("config", () -> new LootFunctionType(new ConfigLootFunction.Serializer()));
}

File: src/main/kotlin/net/hellomouse/xeno_early_start/registries/ProgressionModLootTypeRegistry.kt
package net.hellomouse.xeno_early_start.registries

import net.hellomouse.xeno_early_start.ProgressionMod
import net.hellomouse.xeno_early_start.loot.BlockIsInTagCondition
import net.hellomouse.xeno_early_start.loot.ConfigLootFunction
import net.hellomouse.xeno_early_start.loot.LootTableIdCondition
import net.hellomouse.xeno_early_start.loot.MatchToolTier
import net.minecraft.loot.condition.LootConditionType
import net.minecraft.loot.function.LootFunctionType
import net.minecraft.registry.Registries
import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.RegistryObject
import java.util.function.Supplier

object ProgressionModLootTypeRegistry {
    val COND_DEF_REG: DeferredRegister<LootConditionType> = DeferredRegister.create(
        Registries.LOOT_CONDITION_TYPE.getKey(),
        ProgressionMod.MODID
    )

    @JvmField
    val matchToolTier: RegistryObject<LootConditionType> = COND_DEF_REG.register(
        "match_tool_tier",
        Supplier { LootConditionType(MatchToolTier.Serializer()) })

    @JvmField
    val lootTableIdCondition: RegistryObject<LootConditionType> = COND_DEF_REG.register(
        "loot_table_id",
        Supplier { LootConditionType(LootTableIdCondition.Serializer()) })

    @JvmField
    val blockIsInTagCondition: RegistryObject<LootConditionType> = COND_DEF_REG.register(
        "block_is_in_tag",
        Supplier { LootConditionType(BlockIsInTagCondition.Serializer()) })

    @JvmField
    val FUNC_DEF_REG: DeferredRegister<LootFunctionType> = DeferredRegister.create(
        Registries.LOOT_FUNCTION_TYPE.getKey(),
        ProgressionMod.MODID
    )

    @JvmField
    val configLootFunction: RegistryObject<LootFunctionType> = FUNC_DEF_REG.register(
        "config",
        Supplier { LootFunctionType(ConfigLootFunction.Serializer()) })


}
